demos/object_detection_demo_ssd_async/detection_model_ssd.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "detection_model_ssd.h"
+#include <samples/slog.hpp>
+#include <ngraph/ngraph.hpp>
+
+using namespace InferenceEngine;
+ModelSSD::ModelSSD(const std::string& modelFileName,
+    float confidenceThreshold, bool useAutoResize,
+    const std::vector<std::string>& labels)
+    :DetectionModel(modelFileName, confidenceThreshold, useAutoResize, labels) {
+}
+
+void ModelSSD::onLoadCompleted(InferenceEngine::ExecutableNetwork* execNetwork, RequestsPool* requestsPool)
+{
+    DetectionModel::onLoadCompleted(execNetwork, requestsPool);
+
+    // --- Setting image info for every request in a pool. We can do it once and reuse this info at every submit -------
+    if (inputsNames.size()>1) {
+        for (auto &request : requestsPool->getInferRequestsList()) {
+            auto blob = request->GetBlob(inputsNames[1]);
+            LockedMemory<void> blobMapped = as<MemoryBlob>(blob)->wmap();
+            auto data = blobMapped.as<float *>();
+            data[0] = static_cast<float>(netInputHeight);
+            data[1] = static_cast<float>(netInputWidth);
+            data[2] = 1;
+        }
+    }
+}
+
+std::unique_ptr<ResultBase> ModelSSD::postprocess(InferenceResult& infResult)
+{
+    LockedMemory<const void> outputMapped = infResult.getFirstOutputBlob()->rmap();
+    const float *detections = outputMapped.as<float*>();
+
+    DetectionResult* result = new DetectionResult;
+    *static_cast<ResultBase*>(result) = static_cast<ResultBase&>(infResult);
+
+    auto sz = infResult.metaData.get()->asPtr<ImageMetaData>()->img.size();
+
+    for (size_t i = 0; i < maxProposalCount; i++) {
+        DetectedObject desc;
```

Comment:
I suggest to check image_id and confidence before allocating `DetectedObject`.

Reply:
OK. fixed.